lib/resources/FileUploads.js

```diff
@@ -9,14 +10,38 @@ module.exports = StripeResource.extend({
 
   overrideHost: 'uploads.stripe.com',
 
-  requestDataProcessor: function(method, data, headers) {
+  requestDataProcessor: function(method, data, headers, callback) {
     data = data || {};
 
     if (method === 'POST') {
-      return multipartDataGenerator(method, data, headers);
+      return getProcessorForSourceType(data);
     } else {
       return utils.stringifyRequestData(data);
     }
+
+    function getProcessorForSourceType(data) {
+      var isStream = utils.checkForStream(data);
+      if (isStream) {
+        return streamProcessor(multipartDataGenerator);
+      } else {
+        var buffer = multipartDataGenerator(method, data, headers);
+        return callback(null, buffer);
+      }
+    }
+
+    function streamProcessor (fn) {
+      var bufferArray = [];
+      data.file.data.on('data', function(line) {
+        bufferArray.push(line);
+      }).on('end', function() {
+        var bufferData = Object.assign({}, data);
+        bufferData.file.data = Buffer.concat(bufferArray);
+        var buffer = fn(method, bufferData, headers);
+        callback(null, buffer);
+      }).on('error', function(err) {
+        callback(err, null);
```

Comment:
Per comments above, this is probably the right place to new up the `StreamProcessorError`.

Reply:
updated - thanks!